### DISCUSSION TITLE
Can we support an in memory database for rapid development?

### DISCUSSION THREAD

**Discussion by daninus14** [fukamachi/mito#160] [Ideas]
SQLite supports in memory databases, though I've never tried it.

https://www.sqlite.org/inmemorydb.html

Would it be hard to integrate it into mito? Is it even mito dependent or is this something that has to be done in `cl-dbi`?

I imagine we could have some option `:in-memory` to allow this.

This could be a default for mito if there is no other connection available. We could issue a warning saying defaulting to in-memory sqlite.

I think this would make it easier to explore during development and not have to deal with deleting db files.

**Comment by vindarel** [comment]
We can just do

```lisp
(mito:connect-toplevel :sqlite3
                       :database-name #P":memory:")
```

can't we?

I am against this as a default^^ (and it doesn't make sense anyways, since it isn't an option but we give it instead of a file name).

> I think this would make it easier to explore during development and not have to deal with deleting db files.

A macro for unit tests can be useful. I have one lying around.

**Reply by daninus14** [reply]
Ah thanks! Didn't know we could do this since it's not in the docs! Will add it to the docs now 😄
